spec/pmc_miller_reader_spec.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+require "pmc_miller/reader"
+
+fixtures_dir = File.join("spec", "fixtures", "basic", "puppet-metrics-collector")
+RSpec.describe PmcMiller::Reader do
+  it "allows the setting of the service" do
+    reader = PmcMiller::Reader.new(fixtures_dir)
+    reader.service = :puppetdb
+    expect(reader.service).to eq(:puppetdb)
+  end
+  it "reads service data for given key" do
+    reader = PmcMiller::Reader.new(fixtures_dir)
+    reader.service = :puppetdb
+    data = reader.read("queue_depth")
+    expect(data).to be_an_instance_of(PmcMiller::Data)
```

Comment:
From a user perspective, I think we care more that it actually contain the data from the files than that the type is correct...

Reply:
...which I left as an exercise for the objects using the data.  This is my TDD test for bootstrapping the reader, for which I am only concerned that he correct objects were generated.

Reply:
hm, you mean like, the spec test for quedepth would confirm that the reader actually gave back the right summary and results?  (which I just remembered it does).  That seem backwards actually.  Cause there are more then one analyzers, so doing it that way means they all check that the right data got read?  I would think the reader confirms that data got read, and the analyzers assume it did, and only check that the calculation is correct.